homeassistant/components/sensor/broadlink.py

```diff
@@ -119,8 +118,16 @@ def __init__(self, interval, ip_addr, mac_addr, timeout):
         if not self._auth():
             _LOGGER.warning("Failed to connect to device")
 
+    def _connect(self):
+        import broadlink
+        self._device = broadlink.a1((self.ip_addr, 80), self.mac_addr, None)
+        self._device.timeout = self.timeout
+
     def _update(self, retry=3):
         try:
+            self._connect()
+            if not self._auth():
```

Comment:
This will generate a authorization call for each update

Reply:
Autorisation was not enough, sensor reading still freezes without connect.